src/main/scala/com/kinja/amqp/persistence/MySqlMessageStore.scala

```diff
@@ -6,63 +6,99 @@ import java.text.SimpleDateFormat
 import com.kinja.amqp.model.{ Message, MessageConfirmation }
 
 import scala.concurrent.Future
-import scala.slick.driver.ExtendedProfile
-import scala.slick.jdbc.GetResult.GetLong
-import scala.slick.jdbc.{ GetResult, StaticQuery }
-import scala.slick.lifted.ColumnBase
+import scala.collection.mutable.ListBuffer
 
-abstract class MySqlMessageStore(
-	processId: String,
-	writeDs: javax.sql.DataSource,
-	readDs: javax.sql.DataSource
-) extends MessageStore {
-
-	this: ExtendedProfile =>
-	import simple._
-	import Database.threadLocalSession
-
-	import StaticQuery.interpolation
-
-	val df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+import java.sql.{ Connection, PreparedStatement, ResultSet, Types }
 
-	implicit val getMessageConf = GetResult(r => MessageConfirmation(r.<<, r.<<, r.<<, r.<<, r.<<))
-	implicit val getMessage = GetResult(r => Message(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<))
-
-	object MessageTable extends Table[Message]("rabbit_messages") {
-		def id: Column[Long] = column[Long]("id", O.PrimaryKey, O.AutoInc)
-		def exchangeName: Column[String] = column[String]("exchangeName")
-		def routingKey: Column[String] = column[String]("routingKey")
-		def message: Column[String] = column[String]("message")
-		def channelId: Column[Option[String]] = column[Option[String]]("channelId")
-		def deliveryTag: Column[Option[Long]] = column[Option[Long]]("deliveryTag")
-		def createdTime: Column[Timestamp] = column[Timestamp]("createdTime")
-		def processedBy: Column[Option[String]] = column[Option[String]]("processedBy")
-		def lockedAt: Column[Option[Timestamp]] = column[Option[Timestamp]]("lockedAt")
-
-		def * : ColumnBase[Message] =
-			id.? ~ routingKey ~ exchangeName ~ message ~ channelId ~ deliveryTag ~ createdTime ~ processedBy ~ lockedAt <> (Message.apply _, Message.unapply _)
-
-		def autoInc =
-			id.? ~ routingKey ~ exchangeName ~ message ~ channelId ~ deliveryTag ~ createdTime ~ processedBy ~ lockedAt <> (Message.apply _, Message.unapply _) returning id
+class MySqlMessageStore(
+	processId: String,
+	override val writeDs: javax.sql.DataSource,
+	override val readDs: javax.sql.DataSource) extends MessageStore with ORM {
+
+	private val df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+
+	private val messageFields = List(
+		"id",
+		"routingKey",
+		"exchangeName",
+		"message",
+		"channelId",
+		"deliveryTag",
+		"createdTime",
+		"processedBy",
+		"lockedAt"
+	)
+
+	private val confirmationFields = List(
+		"id",
+		"channelId",
+		"deliveryTag",
+		"multiple",
+		"createdTime"
+	)
+
+	implicit val getMessage = GetResult(r => Message(
+		id = {
+			val value = r.getLong("id")
+			if (r.wasNull) None else Option(value)
+		},
+		routingKey = r.getString("routingKey"),
+		exchangeName = r.getString("exchangeName"),
+		message = r.getString("message"),
+		channelId = {
+			val value = r.getString("channelId")
```

Comment:
I wanted to make sure to handle all the weird JDBC stuff. I'm not convinced whether if the value is NULL in the database, the string will be actually null in the JVM, so I'm calling `.wasNull` to make absolutely sure.